### PR TITLE
tests: add auth unit test

### PR DESCRIPTION
**Pull request by jLopezbarb**
# Proposed changes

- Add tests for the auth graphql flow

### CODE REVIEW

File: pkg/cmd/login/browser.go [hunk 1]
```diff
@@ -47,7 +47,12 @@ func (h *Handler) handle() http.Handler {
 		}
 
 		ctx := r.Context()
-		u, err := okteto.Auth(ctx, code, h.baseURL)
+		oktetoClient, err := okteto.NewOktetoClientFromUrl(h.baseURL)
```
Comment (AdrianPedriza) [inline pkg/cmd/login/browser.go:50]: I think func `NewOktetoClientFromUrl` has enough logic to add some tests for it? :)

File: pkg/okteto/auth.go [hunk 2]
```diff
@@ -66,34 +112,19 @@ func Auth(ctx context.Context, code, url string) (*types.User, error) {
...
-			Token           graphql.String
-			New             graphql.Boolean
-			Registry        graphql.String
-			Buildkit        graphql.String
-			Certificate     graphql.String
-			GlobalNamespace graphql.String  `graphql:"globalNamespace"`
-			Analytics       graphql.Boolean `graphql:"telemetryEnabled"`
-		} `graphql:"auth(code: $code, source: $source)"`
-	}
+	var mutation authMutationStruct
```
Comment (AdrianPedriza) [inline pkg/okteto/auth.go:123]:
```suggestion
	var authMutation authMutationStruct
```
I think thgis improves reading it adding more context

File: pkg/okteto/auth.go [hunk 3]
```diff
@@ -107,46 +138,33 @@ func (c *OktetoClient) authUser(ctx context.Context, code string) (*types.User,
...
-			ExternalID  graphql.String `graphql:"externalID"`
-			Token       graphql.String
-			New         graphql.Boolean
-			Registry    graphql.String
-			Buildkit    graphql.String
-			Certificate graphql.String
-		} `graphql:"auth(code: $code, source: $source)"`
-	}
-
+	var mutation deprecatedAuthMutationStruct
```
Comment (AdrianPedriza) [inline pkg/okteto/auth.go:164]:
```suggestion
	var deprecatedAuthMutation deprecatedAuthMutationStruct
```

File: pkg/okteto/auth.go [hunk 4]
```diff
@@ -42,18 +38,68 @@ type Token struct {
...
+}
+
+func (authenticationErr) Error() string {
+	return "authentication error, please try again"
+}
+func (e authenticationErr) Unwrap() error {
+	return e.E
+}
+
+func newAuthenticationErr(err error) authenticationErr {
```
Comment (AdrianPedriza) [inline pkg/okteto/auth.go:52]: I don't think we need this function, do we? We could use `authenticationErr` directly ?

File: pkg/okteto/context.go [hunk 5]
```diff
@@ -55,6 +56,7 @@ const (
 
 var (
 	CurrentStore *OktetoContextStore
+	reg          = regexp.MustCompile("[^A-Za-z0-9]+")
```
Comment (AdrianPedriza) [inline pkg/okteto/context.go:59]: do we really need this as global var?

File: pkg/cmd/login/browser.go (hunk 1 above)
Comment (AdrianPedriza) [reply pkg/cmd/login/browser.go:50]: NIT: I see this func was added before your changes

**Comment by AdrianPedriza** [comment]
minor comments, LGTM! 👍

**Review by AdrianPedriza — APPROVED** [review]

File: pkg/okteto/context.go (hunk 5 above)
Comment (ifbyol) [reply pkg/okteto/context.go:59]: In that way the regex is compiled one once instead of each time it has to be used

Comment (AdrianPedriza) [reply pkg/okteto/context.go:59]: makes sense!

**Review by ifbyol — APPROVED** [review]

File: pkg/cmd/login/browser.go (hunk 1 above)
Comment (jLopezbarb) [reply pkg/cmd/login/browser.go:50]: That should be addressed when we do the refactor for the `client.go`

**Pull request merged**